engine/alternative-runtimes.md

```diff
@@ -0,0 +1,220 @@
+---
+title: Alternative container runtimes
+description: |
+  Docker Engine uses runc as the default container runtime, but you
+  can specify alternative runtimes using the CLI or by configuring
+  the daemon
+keywords: engine, runtime, containerd, runtime v2, shim
+---
+
+Docker Engine uses containerd for managing the container lifecycle,
+which includes creating, starting, and stopping containers.
+By default, containerd uses runc as its container runtime.
+
+containerd supports a
+[shim API](https://github.com/containerd/containerd/blob/main/runtime/v2/README.md){: target="blank" rel="noopener"}
+which lets you select a different container runtime.
+
+Examples of alternative runtimes that you can use include:
+
+- [youki](https://github.com/containers/youki){: target="blank" rel="noopener"}
+- [Wasmtime](https://wasmtime.dev/){: target="blank" rel="noopener"}
+- [gVisor](https://github.com/google/gvisor){: target="blank" rel="noopener"}
+- [Kata Containers](https://katacontainers.io/){: target="blank" rel="noopener"}
+
+## Use containerd shims
+
+containerd shims let you use alternative runtimes without having to change the
+configuration of the Docker daemon. To use a containerd shim, install the shim
+binary on `PATH` on the system where the Docker daemon is running.
+
+To use a shim with `docker run`, specify the fully qualified name of the
+runtime as the value to the `--runtime` flag:
+
+```console
+$ docker run --runtime io.containerd.kata.v2 hello-world
+```
+
+### Use a containerd shim without installing on PATH
+
+You can use a shim without installing it on `PATH`, in which case you need to
+register the shim in the daemon configuration as follows:
+
+```json
+{
+  "runtimes": {
+    "foo": {
+      "runtimeType": "/path/to/containerd-shim-foobar-v1"
+    }
+  }
+}
+```
+
+To use the shim, specify the name that you assigned to it:
+
+```console
+$ docker run --runtime foo hello-world
+```
+
+### Configure shims
+
+If you need to pass additional configuration for a containerd shim, you can
+use the `runtimes` option in the daemon configuration file.
+
+1. Edit the daemon configuration file by adding a `runtimes` entry for the
+   shim you want to configure.
+
+   - Specify the fully qualified name for the runtime in `runtimeType` key
+   - Add your runtime configuration under the `options` key
+
+   ```json
+   {
+     "runtimes": {
+       "runsc": {
```

Comment:
```suggestion
       "gvisor": {
```
to clarify (somewhat) that runsc is the name of the gVisor runtime and also to demonstrate that the name of the runtime registered with the daemon does not need to have any relation to the runtimeType.

Also: maybe demonstrate that multiple runtimes could be registered using the same runtimeType but different configurations? (Though maybe elsewhere in the document.)
```json
{
  "runtimes": {
    "gvisor-foo": {
      "runtimeType": "io.containerd.runsc.v1",
      "options": {
        "TypeUrl": "io.containerd.runsc.v1.options",
        "ConfigPath": "/etc/containerd/runsc-foo.toml"
      }
    },
    "gvisor-bar": {
      "runtimeType": "io.containerd.runsc.v1",
      "options": {
        "TypeUrl": "io.containerd.runsc.v1.options",
        "ConfigPath": "/etc/containerd/runsc-bar.toml"
      }
    }
  }
}
```

Reply:
Let me add the example of registering the same runtimeType under multiple configurations to the reference doc 👍🏻